subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing_maven.adoc

```diff
@@ -188,6 +188,54 @@ For integration with a local Maven installation, it is sometimes useful to publi
+A custom `message` allows explaining why the artifact coordinates have changed.
+====
+
+The relocation POM should be created for what would be the next version of the old artifact. For example when the artifact coordinates of `com.example:lib:1.0.0` are changed and the artifact with the new coordinates continues version numbering and is published as `com.new-example:lib:2.0.0`, then the relocation POM should specify a relocation from `com.example:lib:**2.0.0**` to `**com.new-example**:lib:2.0.0`.
+
+A relocation POM only has to be published once, the build file configuration for it should be removed again once it has been published.
+
+Note that a relocation POM is not suitable for all situations; when an artifact has been split into two or more separate artifacts then a relocation POM might not be helpful.
+
+[[publishing_maven:retroactive_relocation]]
+=== Retroactively publishing relocation information
+
+It is possible to publish relocation information retroactively after the coordinates of an artifact have changed in the past, and no relocation information was published back then.
+
+The same recommendations as described above apply. To ease migration for users, it is important to pay attention to the `version` specified in the relocation POM. The relocation POM should allow the user to move to the new artifact in one step, and then allow them to update to the latest version in a separate step. For example when for the coordinates of `com.new-example:lib:5.0.0` were changed in version 2.0.0, then ideally the relocation POM should be published for the old coordinates `com.example:lib:**2.0.0**` relocating to `com.new-example:lib:**2.0.0**`. The user can then switch from `com.example:lib` to `com.new-example` and then separately update from version 2.0.0 to 5.0.0, handling breaking changes (if any) step by step.
+
+When relocation information is published retroactively, it is not necessary to wait for next regular release of the project, it can be published in the meantime. As mentioned above, the relocation information should then be removed again from the build file once the relocation POM has been published.
+
+[[publishing_maven:relocation_duplicate_dependencies]]
+=== Avoiding duplicate dependencies
+
+When only the coordinates of the artifact have changed, but package names of the classes inside the artifact have remained the same, dependency conflicts can occur. A project might (transitively) depend on the old artifact but at the same time also have a dependency on the new artifact which both contain the same classes, potentially with incompatible changes.
+
+To detect such conflicting duplicate dependencies, <<dependency_capability_conflict.adoc#sub:declaring-component-capabilities,capabilities>> can be published as part of the <<publishing_gradle_module_metadata.adoc#sec:understanding-gradle-module-md,Gradle Module Metadata>>.
```

Comment:
The [linked section](https://docs.gradle.org/current/userguide/dependency_capability_conflict.html#sub:declaring-component-capabilities) here does not actually explain how do define capabilities for an artifact to be published. Is there a better section describing this, maybe "[Declaring additional capabilities for a local component](https://docs.gradle.org/current/userguide/component_capabilities.html#declaring_additional_capabilities_for_a_local_component)"? (though I am familiar with capabilities)

Additionally how should a project define these capabilities?
Could you please give some hints for how this could be done?
(I might expand this section then)

Reply:
I'll add the link to the additional section, but I don't have any additional hints to provide or link to right now.  Hopefully this will be enough to get someone started.

If you'd like to improve the docs for capabilities please feel free to expand those sections with further examples and submit another PR.